### PR TITLE
Improved aliased parameter parsing

### LINKED ISSUE #3082: Aliased parameters not parsed correctly with apostrophe followed by a whitespace
### Major Version

4.x

### Minor Version Number

3.x & 4.x

### Target environment

All

### Additional environment details

N/A

### Expected or Desired Behavior

The aliased parameters parsing is a very nice feature of this library, but there's an issue when the parameter contains an apostrophe followed by a whitespace.

### Observed Behavior

This is parsed just fine:
`/something('!@p1::value')`

But this gets cut of at the whitespace:
`/something('!@p1::value with'' apostrophe and space')`

### Steps to Reproduce

This is a regex101.com example of the current situation.
https://regex101.com/r/DIigHz/1
image

My suggested fix:
https://regex101.com/r/cgsSEA/1
image

Note the negative lookbehind (`(?<!')`) inserted in the expression.
The only concern here could be [browser support](https://caniuse.com/js-regexp-lookbehind). Depends on which browsers ages you still want to support.

Also. I'm not sure about the `/something('param='!@p1::value'')` example. Is this the actual syntax we would get?
I added it based on this line:
https://github.com/pnp/pnpjs/blob/61df3d1220dd09cc7030534d7db18ff89be24112/packages/sp/spqueryable.ts#L83
Or should it be `/something(param='!@p1::value')`?

### PR DESCRIPTION
**Pull request by RoelVB**
### Category

- [x] Bug fix?
- [ ] New feature?
- [ ] New sample?
- [ ] Documentation update?

### Related Issues

fixes #3082

### What's in this Pull Request?

This PR improves aliased parameter parsing, includes all sensible tests I could think of.

### CODE REVIEW

File: packages/sp/spqueryable.ts [hunk 1]
```diff
@@ -75,17 +75,11 @@ export class _SPQueryable<GetType = any> extends Queryable<GetType> {
...
-        // /something(!@p1::value)
-        // /something(!@p1::value, param=value)
-        // /something(param=value,!@p1::value)
-        // /something(param=value,!@p1::value,param=value)
-        // /something(param=!@p1::value)
-        // there could be spaces or not around the boundaries
-        let url = this.toUrl().replace(/([( *| *, *| *= *])'!(@.*?)::(.*?)'([ *)| *, *])/ig, (match, frontBoundary, labelName, value, endBoundary) => {
+        // this regex is designed to locate aliased parameters within url paths
+        let url = this.toUrl().replace(/'!(@.+?)::((?:[^']|'')+)'/ig, (match, labelName, value) => {
             this.log(`Rewriting aliased parameter from match ${match} to label: ${labelName} value: ${value}`, 0);
```
Comment (juliemturner) [inline packages/sp/spqueryable.ts:80]: Did you intend to leave this logging in the final solution?

Comment (RoelVB) [reply packages/sp/spqueryable.ts:80]: You mean the ``this.log(`Rewriting aliased parameter from match...`` line?
It was already there, so I kept it.

Comment (juliemturner) [reply packages/sp/spqueryable.ts:80]: Ahh ok, probably an oversight, we'll remove it.

**Comment by RoelVB** [comment]
Am I correct to assume all tests with a dash (`-`) in front are skipped? Those are quite a few, including the new one

**Comment by juliemturner** [comment]
@RoelVB - I'm not sure what you mean... tests are skipped when they are declared `it.skip(...` or they use the `this.skip()` call in the test body.

**Comment by RoelVB** [comment]
@juliemturner Sorry for the very unclear comment.
I mean this [in the action](https://github.com/pnp/pnpjs/actions/runs/9891994887/job/27328087375?pr=3084):
image
Those "pending" include my new test

**Review by juliemturner — APPROVED** [review]

**Pull request merged**